_drgn.pyi

```diff
@@ -448,7 +450,39 @@ class Program:
         :param fn: Callable taking a program, name, :class:`FindObjectFlags`,
             and filename: ``(prog, name, flags, filename)``. The filename
             should be matched with :func:`filename_matches()`. This should
-            return an :class:`Object`.
+            return an :class:`Object` or ``None`` if not found.
+        """
+        ...
+    def add_symbol_finder(
+        self, fn: Callable[[Optional[str], Optional[int], bool], List[Symbol]]
```

Comment:
We should allow this to return any `Sequence[Symbol]` instead of specifically a `List[Symbol]`. See [`PySequence_Fast`](https://docs.python.org/3/c-api/sequence.html#c.PySequence_Fast) for one way to implement that. We use that in a few places in the Python bindings already.